x/precisebank/keeper/view.go

```diff
@@ -0,0 +1,37 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/kava-labs/kava/x/precisebank/types"
+)
+
+// GetBalance returns the balance of a specific denom for an address. This will
+// return the extended balance for the ExtendedCoinDenom, and the regular
+// balance for all other denoms.
+func (k Keeper) GetBalance(
+	ctx sdk.Context,
+	addr sdk.AccAddress,
+	denom string,
+) sdk.Coin {
+	// Pass through to x/bank for denoms except ExtendedCoinDenom
+	if denom != types.ExtendedCoinDenom {
+		return k.bk.GetBalance(ctx, addr, denom)
+	}
+
+	// x/bank for integer balance - spendable balance only
+	spendableCoins := k.bk.SpendableCoins(ctx, addr)
```

Comment:
i see that x/bank's keeper's GetBalance returns total balance including locked coins.

why not the true balance here? is it to avoid complications with vesting accounts? or do we only return spendable balance b/c there's no concept of non-spendable balance in evmland?

Reply:
right hmm this does diverge from `x/bank` behavior which is not great. and yeah I think it was originally used in evmutil to avoid having vested coins show up in balances since that would be confusing to get insufficient balance errors when the queried balance is enough.